### PR TITLE
SONARIAC-384 S6317: Update message and supports secondary locations

### PR DESCRIPTION
**Pull request by navendu-barua-sonarsource**

### CODE REVIEW

**Review by karim-ouerghemmi-sonarsource — CHANGES_REQUESTED** [review]
I did leave some comments we can work on.

File: iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationVector.java [hunk 1]
```diff
@@ -21,51 +21,70 @@
...
+  public static Optional<PrivilegeEscalationVector> getEscalationVector(Stream<String> actionPermissions) {
     Set<Permission> permissionVector = actionPermissions.map(Permission::of).collect(Collectors.toSet());
-    return Stream.of(PrivilegeEscalationVector.values()).anyMatch(vector -> vector.isSubsetOf(permissionVector));
+
+    return Stream.of(PrivilegeEscalationVector.values())
+      .filter(vector -> vector.isSubsetOf(permissionVector))
+      .findFirst();
+  }
+
+  public List<Permission.SimplePermission> permissions() {
```
Comment (karim-ouerghemmi-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationVector.java:82]: Let's also rename this getter to `getPermissions` to align with the name getter.

**Review by karim-ouerghemmi-sonarsource — APPROVED** [review]
LGTM. Nice!
left two very minor comments.

File: iac-extensions/cloudformation/src/main/java/org/sonar/iac/cloudformation/checks/PrivilegeEscalationCheck.java [hunk 2]
```diff
@@ -66,34 +67,32 @@ private static void checkPrivilegeEscalation(CheckContext ctx, Policy policy, Re
...
     return secondaryLocations;
   }
 
-  private static Optional<PrivilegeEscalationVector> getPrivilegeEscalationVector(Statement statement) {
+  private static boolean actionEnablesVector(PrivilegeEscalationVector vector, String value) {
+    PrivilegeEscalationVector.Permission permission = PrivilegeEscalationVector.Permission.of(value);
+    return vector.permissions().stream().anyMatch(p -> p.isCoveredBy(permission));
+  }
+
+  private static Optional<PrivilegeEscalationVector> getStatementEscalationVecotr(Statement statement) {
```
[the remark this answers is not in the document]
Comment (navendu-barua-sonarsource) [reply iac-extensions/cloudformation/src/main/java/org/sonar/iac/cloudformation/checks/PrivilegeEscalationCheck.java:88]: done

File: iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationVector.java (hunk 1 above)
Comment (navendu-barua-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationVector.java]: done

**Pull request merged**